github-comment-fetcher/src/main/java/com/tzachz/commentcounter/PullRequestCache.java

```diff
@@ -18,6 +18,7 @@
 public class PullRequestCache {
 
     public static final GHPullRequest UNKNOWN = new GHPullRequest(new GHUser("UNKNOWN", ""));
+    public static final GHPullRequest NOT_FOUND = new GHPullRequest(null);
```

Comment:
@tzachz 
Instead of using `Optional`, how do you feel about creating an exception class for when a GitHub resource is not found and throwing that instead? (This `PullRequestCache` might need another set or cache to hold URLs that are not found)